content/automate/ManageIQ/System/Event/StateMachines/Refresh.class/__methods__/check_refreshed.rb

```diff
@@ -0,0 +1,63 @@
+#
+# Description: This method checks the refresh completion.
+#
+
+module ManageIQ
+  module Automate
+    module System
+      module Event
+        module StateMachines
+          module Refresh
+            class CheckRefreshed
+              STATE_FINISHED = 'Finished'.freeze
+
+              def initialize(handle = $evm)
+                @handle = handle
+              end
+
+              def main
+                check_status(refresh_task)
+              end
+
+              private
+
+              def check_status(task)
+                case task.state
+                when STATE_FINISHED
```

Comment:
@lfu What about the error scenario when the task is finished but has an error? will that get flagged as a retry

Reply:
@mkanoor A ```MiqTask's```  state may be one of ```Initialized, Queued, Active and Finished```.

Reply:
@mkanoor There is a separate `status` column for capturing that.  [app/models/miq_task.rb#L8-L12](https://github.com/ManageIQ/manageiq/blob/d992f980a857db789c8aa8b3636bf1b6559ca539/app/models/miq_task.rb#L8-L12)